tests/src/Context/MinkContext.php

```diff
@@ -86,4 +89,108 @@ public function iWaitForAjaxToFinish($event = NULL) {
     parent::iWaitForAjaxToFinish($event);
   }
 
+  /**
+   * {@inheritdoc}
+   *
+   * Overrides the parent method in order to support Select2.
+   */
+  public function selectOption($select, $option): void {
+    if ($field = $this->select2IsUsed($select, $option)) {
+      $xpath = '//li[contains(@class, "select2-selection__choice")]/span[contains(@class, "select2-selection__choice__remove")]';
+      if ($selected_remove_buttons = $field->getParent()->findAll('xpath', $xpath)) {
+        // This method selects anew an option in a normal select list. To have
+        // the same results in a select2 widget, all selected options have to
+        // be removed manually.
+        // To select an additional item, use the ::additionallySelectOption
+        // method.
+        foreach ($selected_remove_buttons as $selected_remove_button) {
```

Comment:
Great catch! I fully agree with your point, the 1st method, `::selectOption()`, is meant to do a single selection, like doing a simple click on a multi-value select. That meaning all options previously selected are deselected. While `::additionallySelectOption()` would append new selections to the already selected options, simulating holding the CTRL key.